pyhf/pdf.py

```diff
@@ -131,6 +132,17 @@ def __init__(self, spec, **config_kwargs):
         # build up our representation of the specification
         self.config = _ModelConfig.from_spec(self.spec,**config_kwargs)
 
+
+        for m in self.config.modifiers:
```

Comment:
probably require a `def finalize(self): pass` on all other modifiers to be honest.

Reply:
hm not sure, so far we only have a single one that needs finalization. I'm not sure we will see more

Reply:
that's probably a good point. Can't the `finalize` in staterror just be part of the initialization and it provides a `self.sigmas` that you need to extract?